client/src/pages/CataloguePage/CataloguePage.js

```diff
@@ -16,29 +19,62 @@ const FilterContainer = styled((props) => <Grid item xs={12} {...props} />)(({ t
   },
 }));
 
-const CataloguePage = () => (
-  <Box sx={{ backgroundColor: '#EDEDED', paddingBottom: '150px' }}>
-    <CatalogMainSection />
-    <Container>
-      <Grid container sx={{ mt: '60px', gap: '40px' }}>
-        <Grid item xs={12} laptop sx={{ p: 0 }}>
-          <Typography variant="h2" sx={{ textTransform: 'uppercase', mb: '25px' }}>
-            Tours
-          </Typography>
-          <Stack spacing={2}>
-            <CatalogTourCard />
-            <CatalogTourCard />
-            <CatalogTourCard />
-            <CatalogTourCard />
-          </Stack>
-        </Grid>
+const CataloguePage = () => {
+  const dispatch = useDispatch();
+  const [loading, setLoading] = useState(null);
+  const products = useSelector((state) => state.catalogueReducer.products, shallowEqual);
+  useEffect(() => {
+    setLoading(true);
+    const getData = async () => {
+      const data = await getProducts();
+      dispatch(setProducts(data));
+      setLoading(false);
+    };
+    if (products.length > 0) {
+      setLoading(false);
+    } else {
+      getData();
+    }
+  }, []);
```

Comment:
this is not correct,   we are using Redux toolkit.  the async request has to be encapuslated in the Slice file.  and just called in here.    you can check my tourSlice as an example. you will find a THUNK function called fetchTour

isLoading logic also has to be encapsulated into the Redux thunk function.